sympy/ntheory/factor_.py

```diff
@@ -237,6 +237,9 @@ def multiplicity(p, n):
     if p == n:
         return 1
 
+    if n == 0:
+        raise ValueError('n cannot be zero, but got %s' % n)
```

Comment:
Also since with this code `multiplicity(2, 0)` is `0`.  So the position of `if` conditional should be a little above this.